File: src/output.ts
export function printResult(data: unknown, humanText: string, json: boolean): void {
  if (json) {
    process.stdout.write(`${JSON.stringify(data, null, 2)}\n`);
  } else {
    process.stdout.write(`${humanText}\n`);
  }
}

export function printError(message: string, json: boolean): void {
  if (json) {
    process.stderr.write(`${JSON.stringify({ error: message }, null, 2)}\n`);
  } else {
    process.stderr.write(`Error: ${message}\n`);
  }
}

File: python/src/computeledger/output.py
"""Human-readable vs. structured JSON output, matching ``src/output.ts``.

Note: unlike :mod:`computeledger.canonical`, this module's JSON output is
never hashed or signed — it exists purely for human/agent consumption — so
it uses Python's regular ``json.dumps`` pretty-printing rather than the
byte-exact canonical serializer.
"""

from __future__ import annotations

import json
import sys
from typing import Any


def print_result(data: Any, human_text: str, json_output: bool) -> None:
    if json_output:
        sys.stdout.write(json.dumps(data, indent=2, ensure_ascii=False) + "\n")
    else:
        sys.stdout.write(human_text + "\n")


def print_error(message: str, json_output: bool) -> None:
    if json_output:
        sys.stderr.write(json.dumps({"error": message}, indent=2, ensure_ascii=False) + "\n")
    else:
        sys.stderr.write(f"Error: {message}\n")
